packages/cli-kit/src/public/node/themes/types.ts

```diff
@@ -82,6 +83,15 @@ export interface ThemeFileSystem {
   addEventListener: {
     <T extends ThemeFSEventName>(eventName: T, cb: (params: ThemeFSEventPayload<T>) => void): void
   }
+
+  /**
+   * Starts a file watcher for the theme directory.
+   *
+   * @param themeId - The ID of the theme being watched.
+   * @param adminSession - The admin session for API communication.
+   * @returns A Promise that resolves to an FSWatcher instance.
+   */
+  startWatcher: (themeId: string, adminSession: AdminSession) => Promise<void>
```

Comment:
This allows us to provide the `themeId` and `adminSession` without modifying `mountThemeFileSystem`
- I wanted to separate this logic since we are using the local theme file system in commands like `push` where we don't need watching behaviour
- The downside of this approach is that you need to explicitly call `startWatcher` - simply adding a listener is not enough

Another alternative is to include  `themeId` and `adminSession` as optional args in `mountThemeFileSystem`

```
export async function mountThemeFileSystem(root: string, themeId?: string, adminSession?: AdminSession): Promise<ThemeFileSystem> {
....existing code

if (themeId && adminSesssion) {
const watcher = chokidar.watch([...directoriesToWatch], {
        ignored: THEME_DEFAULT_IGNORE_PATTERNS,
        persistent: !process.env.SHOPIFY_UNIT_TEST,
        ignoreInitial: true,
      })

      watcher
        .on('add', handleFileUpdate.bind(null, 'add', themeId, adminSession))
        .on('change', handleFileUpdate.bind(null, 'change', themeId, adminSession))
        .on('unlink', handleFileDelete.bind(null, themeId, adminSession))
}
```

I've considered adding `themeId` and `adminSession` as args in `addEventListener` but that felt smelly - I wouldn't expect to need to provide those arguments each time I call `addEventListener`